core/commands/commit.py

```diff
@@ -57,11 +59,37 @@ class GsCommitInitializeViewCommand(TextCommand, GitCommand):
     the previous commit message if amending.
     """
 
+    def amending(self):
+        return self.view.settings().get("git_savvy.commit_view.amend")
+
+    def fetch_commit_summary(self):
+        """
+        Fetch the same pre-commit template git would use if you were doing
+        `git commit` (showing staged, unstaged, etc.)
+        """
+
+        args = []
+        if self.amending():
+            args.append("--amend")
+
+        dryrun = self.git("commit", "--dry-run", *args, throw_on_stderr = False)
+        # detect trying to do a commit with nothing staged
+        if dryrun.find('no changes added') != -1:
+            sublime.error_message(NO_STAGED_FILES)
```

Comment:
I wasn't sure what the proper thing to do was since at that point we already have the view (it's created by the first command at the window level).